templates/tapestry.js

```diff
@@ -2021,20 +2017,30 @@ function tapestryTool(config){
     
     /* For setting the "unlocked" field of nodes in dataset if logic shows node to be unlocked */
     function setUnlocked() {
-        // this is here because the other unlock fix wasn't working for some reason.
-        tapestry.dataset.nodes.forEach(node => {
-            node.unlocked = true
-        });
-        /* var parentIndex;
-        for (var i = 0; i < tapestry.dataset.links.length; i++) {
-            
-            childIndex = findNodeIndex(tapestry.dataset.links[i].target.id);
-            parentIndex = findNodeIndex(tapestry.dataset.links[i].source.id);
-    
-            if (tapestry.dataset.links[i].appearsAt <= (tapestry.dataset.nodes[parentIndex].typeData.progress[0].value * tapestry.dataset.nodes[parentIndex].mediaDuration)) {
-                tapestry.dataset.nodes[childIndex].unlocked = true;
-            }
-        } */
+        const conditionTypes = {
+            NODE_COMPLETED: "node_completed",
+        }
+
+        const { nodes } = tapestry.dataset
+        nodes.forEach(node => {
+            const { conditions } = node
+            conditions.forEach(condition => {
+                const dependency = nodes[findNodeIndex(condition.value)]
+                let nodeDependents = dependency.dependents
+                nodeDependents.push({id: node.id, condition: condition})
+                dependency.dependents = nodeDependents
```

Comment:
@ItsAdel I think I know what's causing the duplicate conditions. `setUnlocked` is called every time the tapestry rerenders, and every time the nodeDependents list gets updated. Maybe add an if statement before to ensure the condition doesn't exist already? Or change line to:

```suggestion
                let nodeDependents = []
                nodeDependents.push({id: node.id, condition: condition})
                dependency.dependents = nodeDependents
```
Although this would cause a new array to be created every time, which might be a bit inefficient.

Reply:
Nevermind disregard this - I found a workaround that doesn't involve reinitializing the entire tapestry, so this doesn't happen anymore.